### ISSUE TITLE
"Ignore Holidays" list leads to "User input malformed"

### ISSUE THREAD

**Issue by rccoleman** [bruxy70/Garbage-Collection#278]
**Describe the bug**

I'm in the US in California and have an "Ignore holiday" list that looks like this:
`Columbus Day,Veterans Day,Martin Luther King Jr. Day,Washington's Birthday,César Chávez Day,Susan B. Anthony Day`

I added this quite a while ago, but as I was modifying some other parameters in that config page (the second one), I noticed that I get "User input malformed" if I try to submit that page with the pre-filled input in that field in place.  The only way I can edit and successfully submit anything on that page is if I cut/remove that whole string and just paste it right back.  Hitting "submit" when I do that works fine.  I'm just cutting and pasting exactly what was already there, so I don't know what's causing it to think the input is malformed.

**Configuration**

Do you configure the integration in YAML or Config Flow (GUI)? :  Config Flow

This is the config page as I land on it after hitting "submit" on the first page:

<img width="603" alt="Screen Shot 2021-07-10 at 9 22 11 AM" src="https://user-images.githubusercontent.com/860888/125169763-7e7a8200-e160-11eb-90c5-eed72194212b.png">

And if I just hit "submit" immediately without changing anything:

<img width="606" alt="Screen Shot 2021-07-10 at 9 22 27 AM" src="https://user-images.githubusercontent.com/860888/125169775-8d613480-e160-11eb-9394-8ca376f28953.png">

Simply cutting and pasting the "ignore list" allows the submit to proceed.

I have the following in my HA log configuration and I see no errors when this happens (I do see the rest of the debug):

```
    custom_components.garbage_collection: debug
    holidays: debug
```

**Error log/debug information**

There are no errors reported in the HA log when I get "User input malformed", even with debug enabled.

**Screenshots**

**Version**

Integration version: 3.2.0

Home Assistant version: 2021.7.1

**Comment by bruxy70** [comment]
I think it has to do with the ` in one of the strings. Will check that.

**Comment by rccoleman** [comment]
I wouldn't be surprised if that was it. Unfortunately it has to match the name that comes from the holiday list punctuation and all.

**Comment by bruxy70** [comment]
Will be fixed by #287
